### ISSUE TITLE
Panic when setting --query-queue-size to a large value

### ISSUE THREAD

**Issue by danxmoran** [influxdata/influxdb#20145]
__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Run `influxd --query-queue-size 9223372036854775807`

__Expected behavior:__
Either the program runs successfully, or the CLI errors out telling me the number is too big.

__Actual behavior:__
```
panic: makechan: size out of range

goroutine 1 [running]:
github.com/influxdata/influxdb/v2/query/control.New(0xa, 0x0, 0x7fffffffffffffff, 0x0, 0x7fffffffffffffff, 0xc001a4c2a0, 0x0, 0x0, 0x0, 0xc000dc2a60, ...)
        /Users/dan/influxdb/query/control/controller.go:180 +0x605
github.com/influxdata/influxdb/v2/cmd/influxd/launcher.(*Launcher).run(0xc00042d880, 0x760eb80, 0xc0009ae0c0, 0x0, 0x0)
        /Users/dan/influxdb/cmd/influxd/launcher/launcher.go:901 +0x3a38
github.com/influxdata/influxdb/v2/cmd/influxd/launcher.cmdRunE.func1(0x0, 0x0)
        /Users/dan/influxdb/cmd/influxd/launcher/launcher.go:149 +0x14c
github.com/influxdata/influxdb/v2/kit/cli.NewCommand.func1(0xc000218b00, 0xc000c923a0, 0x0, 0x2, 0x0, 0x0)
        /Users/dan/influxdb/kit/cli/viper.go:62 +0x29
github.com/spf13/cobra.(*Command).execute(0xc000218b00, 0xc00004e1c0, 0x2, 0x2, 0xc000218b00, 0xc00004e1c0)
        /Users/dan/.gvm/pkgsets/go1.15/global/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0xc000218b00, 0xc000d1ff48, 0x1, 0x1)
        /Users/dan/.gvm/pkgsets/go1.15/global/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/dan/.gvm/pkgsets/go1.15/global/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
        /Users/dan/influxdb/cmd/influxd/main.go:41 +0x31c
```

__Environment info:__
InfluxDB OSS 2.0.2

**Comment by danxmoran** [comment]
Experimenting in the Go playground, it seems that `35184372088820` is the max size that avoids a panic. This is `1<<45-12`, the same as the max size for a `chan` of `int64`s.

While it doesn't panic, it immediately OOMs with no other changes to settings.

**Comment by danxmoran** [comment]
Closed by #20149
